src/main/scala/com/databricks/labs/mosaic/core/Mosaic.scala

```diff
@@ -1,21 +1,62 @@
 package com.databricks.labs.mosaic.core
 
-import scala.annotation.tailrec
-
 import com.databricks.labs.mosaic.core.geometry.MosaicGeometry
 import com.databricks.labs.mosaic.core.geometry.api.GeometryAPI
 import com.databricks.labs.mosaic.core.geometry.linestring.MosaicLineString
 import com.databricks.labs.mosaic.core.geometry.multilinestring.MosaicMultiLineString
+import com.databricks.labs.mosaic.core.geometry.multipoint.MosaicMultiPoint
+import com.databricks.labs.mosaic.core.geometry.point.MosaicPoint
 import com.databricks.labs.mosaic.core.index.IndexSystem
 import com.databricks.labs.mosaic.core.types.model.{GeometryTypeEnum, MosaicChip}
-import com.databricks.labs.mosaic.core.types.model.GeometryTypeEnum.{LINESTRING, MULTILINESTRING}
+import com.databricks.labs.mosaic.core.types.model.GeometryTypeEnum._
+
+import scala.annotation.tailrec
 
 /**
   * Single abstracted logic for mosaic fill via [[IndexSystem]]. [[IndexSystem]]
   * is in charge of implementing the individual steps of the logic.
   */
 object Mosaic {
 
+    def getChips(
+        geometry: MosaicGeometry,
+        resolution: Int,
+        keepCoreGeom: Boolean,
+        indexSystem: IndexSystem,
+        geometryAPI: GeometryAPI
+    ): Seq[MosaicChip] = {
+        GeometryTypeEnum.fromString(geometry.getGeometryType) match {
+            case POINT           => pointChip(geometry, resolution, keepCoreGeom, indexSystem)
+            case MULTIPOINT      => multiPointChips(geometry, resolution, keepCoreGeom, indexSystem)
+            case LINESTRING      => lineFill(geometry, resolution, indexSystem, geometryAPI)
```

Comment:
Linestring andd Multilinestring both have another switch like this.
Is it worth refactoring this class to separate them out and rename all methods like lineChips, multil;ineChips, polygonChips?